pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -164,7 +164,7 @@ ctypedef fused vector_data:
     StringVectorData
 
 cdef bint needs_resize(vector_data *data) noexcept nogil:
-    return data.n == data.m
+    return data.size == data.capacity
```

Comment:
I don't know if `size` ever increments more than 1 before checking, but a `>=` would probably be safer here.